scripts/runnable_recipe_ray_vllm_weight_sync.py

```diff
@@ -368,6 +369,447 @@ def run(self):
             idx += 1
 
 
+# =============================================================================================
+
+from typing import Dict, Any
+from vllm import LLM, SamplingParams
+import vllm
+import torch
+from tensordict import TensorDictBase, TensorClass, TensorDict, NonTensorStack, lazy_stack, from_dataclass
+from tensordict.utils import _zip_strict, expand_as_right
+
+# class Trajectory(TensorClass):
+#     text: str
+#     tokens: torch.Tensor|None=None
+#     response: str | None=None
+#     # response_tokens --> tokens_response because Transform hardcodes..
+#     tokens_response: torch.Tensor | None=None
+#     # prompt_attention_mask: torch.Tensor | None=None
+#     log_probs: torch.Tensor | None=None
+
+def vllm_generate(
+    tensordict: TensorDictBase | Trajectory,
+    vllm_instance,  # The LLM object
+    generate_kwargs: Dict[str, Any] | None = None,
+    tokenizer = None,
+    text_key: str = "text",
+    token_key: str = "tokens",
+    # response_tokens --> tokens_response because Transform hardcodes..
+    token_response_key: str = "tokens_response",
+    text_response_key: str = "text_response",
+    log_prob_key: str = "log_probs",
+    attention_mask_key: str = "attention_mask",
+    padding_value: int = -1,
+) -> Trajectory:
```

Comment:
You are returning a Trajectory, but the definition is commented out?

Reply:
This is now returning a tensordict


If we return a trajectory here this would break with the rest of the torchrl code that accesses attributes with square brackets.

Discussed this with @vmoens and we can put a Trajectory in the queue